x/stakeibc/keeper/msg_server_rebalance_validators.go

```diff
@@ -0,0 +1,99 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/Stride-Labs/stride/x/stakeibc/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	stakingTypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+func abs(n int64) int64 {
+	if n < 0 {
+		return -n
+	}
+	return n
+}
+
+func (k msgServer) RebalanceValidators(goCtx context.Context, msg *types.MsgRebalanceValidators) (*types.MsgRebalanceValidatorsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	hostZone, found := k.GetHostZone(ctx, msg.HostZone)
+	if !found {
+		k.Logger(ctx).Info(fmt.Sprintf("Host Zone not found %s", msg.HostZone))
+		return nil, types.ErrInvalidHostZone
+	}
+
+	validatorDeltas, err := k.GetValidatorAmtDifferences(ctx, hostZone)
+	if err != nil {
+		k.Logger(ctx).Info(fmt.Sprintf("Error getting validator deltas for Host Zone %s: %s", hostZone.ChainId, err))
+		return nil, err
+	}
+
+	// we convert the above map into a list of tuples
+	type valPair struct {
+		deltaAmt int64
+		valAddr  sdk.ValAddress
+	}
+	valDeltaList := make([]valPair, len(validatorDeltas))
+	for valAddr, deltaAmt := range validatorDeltas {
+		valDeltaList = append(valDeltaList, valPair{deltaAmt, sdk.ValAddress(valAddr)})
+	}
+	// now we sort that list
+	lessFunc := func(i, j int) bool {
+		return valDeltaList[i].deltaAmt < valDeltaList[j].deltaAmt
+	}
+	sort.Slice(valDeltaList, lessFunc)
+	// now varDeltaList is sorted by deltaAmt
+	underWeightIndex := 0
+	overWeightIndex := len(valDeltaList) - 1
+
+	var msgs []sdk.Msg
+	delegatorAddressStr := hostZone.DelegationAccount.Address
+	delegatorAddress := sdk.AccAddress(delegatorAddressStr)
+
+	for i := 1; i < 6; i++ {
+		underWeightElem := valDeltaList[underWeightIndex]
+		overWeightElem := valDeltaList[overWeightIndex]
+		if (underWeightElem.deltaAmt < 0) == (overWeightElem.deltaAmt < 0) {
```

Comment:
I think there's another invariant here which is the underWeightElem *must* be <= 0 and the overWeightElem *must* be >= 0, wdyt of explicitly checking this?